internal/batches/executor/coordinator.go

```diff
@@ -254,37 +260,55 @@ func (c *Coordinator) cacheAndBuildSpec(ctx context.Context, taskResult taskResu
 	return specs, nil
 }
 
-// Execute executes the given Tasks and the importChangeset statements in the
-// given spec. It regularly calls the executionProgressPrinter with the
-// current TaskStatuses.
-func (c *Coordinator) Execute(ctx context.Context, tasks []*Task, spec *batcheslib.BatchSpec, ui TaskExecutionUI) ([]*batcheslib.ChangesetSpec, []string, error) {
-	var (
-		specs []*batcheslib.ChangesetSpec
-		errs  *multierror.Error
-	)
+// Execute executes the given tasks. It calls the ui on updates.
+func (c *Coordinator) Execute(ctx context.Context, tasks []*Task, ui TaskExecutionUI) error {
+	results, err := c.doExecute(ctx, tasks, ui)
 
-	ui.Start(tasks)
-
-	// Run executor
-	c.exec.Start(ctx, tasks, ui)
-	results, err := c.exec.Wait(ctx)
-	if err != nil {
-		if c.opts.SkipErrors {
-			errs = multierror.Append(errs, err)
-		} else {
-			return nil, nil, err
+	// Write results to cache.
+	for _, taskResult := range results {
+		if cacheErr := c.writeCache(ctx, taskResult, ui); cacheErr != nil {
+			return cacheErr
 		}
 	}
 
+	return err
+}
+
+// ExecuteAndBuildSpecs executes the given tasks and builds changeset specs for the results.
+// It calls the ui on updates.
+func (c *Coordinator) ExecuteAndBuildSpecs(ctx context.Context, batchSpec *batcheslib.BatchSpec, tasks []*Task, ui TaskExecutionUI) ([]*batcheslib.ChangesetSpec, []string, error) {
+	results, errs := c.doExecute(ctx, tasks, ui)
+
+	var specs []*batcheslib.ChangesetSpec
+
 	// Write results to cache, build ChangesetSpecs if possible and add to list.
 	for _, taskResult := range results {
-		taskSpecs, err := c.cacheAndBuildSpec(ctx, taskResult, ui)
+		taskSpecs, err := c.writeCacheAndBuildSpecs(ctx, batchSpec, taskResult, ui)
 		if err != nil {
 			return nil, nil, err
 		}
 
 		specs = append(specs, taskSpecs...)
 	}
 
-	return specs, c.logManager.LogFiles(), errs.ErrorOrNil()
+	return specs, c.logManager.LogFiles(), errs
+}
+
+func (c *Coordinator) doExecute(ctx context.Context, tasks []*Task, ui TaskExecutionUI) (results []taskResult, err error) {
+	var errs *multierror.Error
+
+	ui.Start(tasks)
+
+	// Run executor
+	c.exec.Start(ctx, tasks, ui)
+	results, err = c.exec.Wait(ctx)
+	if err != nil {
+		if c.opts.SkipErrors {
```

Comment:
Not sure if it's new behaviour but I think errors should be reported in the UI in any case, right?